keras_cv/models/feature_extractor/coca/coca_model.py

```diff
@@ -0,0 +1,311 @@
+            self.encoder_width, self.img_patch_size
+        )
+        self.image_encoder = Sequential(
+            [
+                CVTransformerEncoder(
+                    self.encoder_width,
+                    self.encoder_heads,
+                    self.encoder_intermediate_dim,
+                )
+                for _ in range(self.encoder_depth)
+            ]
+        )
+
+        self.text_embedding = RotaryEmbedding()
+        self.unimodal_text_decoder = Sequential(
+            [
+                TransformerDecoder(
+                    self.decoder_intermediate_dim, self.unimodal_decoder_heads
+                )
+                for _ in range(self.unimodal_decoder_depth)
+            ]
+        )
+        self.multimodal_text_decoder = Sequential(
+            [
+                TransformerDecoder(
+                    self.decoder_intermediate_dim, self.multimodal_decoder_heads
+                )
+                for _ in range(self.multimodal_decoder_depth)
+            ]
+        )
+
+        self.contrastive_attn_pooling = AttentionPooling(
+            self.encoder_width, self.contrastive_attn_heads
+        )
+        self.captioning_attn_pooling = AttentionPooling(
+            self.encoder_width, self.captioning_attn_heads
+        )
+
+        # These are learnable weights defined in build as per Keras recommendations
+        self.cls_token = None
+        self.contrastive_query = None
+        self.captioning_query = None
+
+    def build(self, input_shape):
+        super().build(input_shape)
+
+        # Validate Input Shape
+        if len(input_shape) < 2:
+            raise ValueError(
+                "Build arguments to coca expected to contain shapes of both text and image data; "
+                f"got {len(input_shape)} shapes."
+            )
+
+        images_shape = input_shape[0]
+        text_shape = input_shape[1]
+
+        if len(images_shape) != 4:
+            raise ValueError(
+                "Image shape expected to be of shape [batch_size, height, width, channels]. Instead got "
+                f"shape: {images_shape}"
+            )
+        elif len(text_shape) != 2:
+            raise ValueError(
+                "Text shape expected to be of shape [batch_size, context_length]. Instead got shape"
+                f": {text_shape}"
+            )
+
+        text_dim = text_shape[1]
+        batch_size = images_shape[0]
+        if batch_size != text_shape[0]:
+            raise ValueError(
+                f"Differing batch sizes between images and texts input. {batch_size} vs {text_shape[0]}"
+            )
+
+        # Build Layers
+        self.image_patching.build(images_shape)
+
+        # Add 1 for CLs token appended by patching
+        num_patches = (images_shape[1] // self.img_patch_size) * (
+            images_shape[2] // self.img_patch_size
+        ) + 1
+        self.image_encoder.build((batch_size, self.encoder_width, num_patches))
```

Comment:
you could keep batch_size as `None`
example 
`self.image_encoder.build((None, self.encoder_width, num_patches))`

Reply:
Just for my understanding, is there a specific reason to do that over setting the batch_size?